flowauth/backend/flowauth/models.py

```diff
@@ -271,13 +270,14 @@ def validate(self, code: str) -> bool:
         is_valid = pyotp.totp.TOTP(self.decrypted_secret_key).verify(code)
         if is_valid:
             if (
-                self.last_used_two_factor_code == code
+                current_app.config["CACHE_BACKEND"].get(f"{self.user_id}".encode())
```

Comment:
Can there ever be more than one valid code at a time (excluding backup codes)? If so, then codes could be re-used by alternating between two valid codes, since this only checks the last-used code. But as long as there's only one valid code at a time then I think this is fine.

Reply:
We don't change the default `valid_window=0` when `verify`ing the code, so there's only one valid code at a time.